### PR TITLE
Resolves #117: add `.env` support

### LINKED ISSUE #117: Feature Request: Read sensitive keys from the environment / .env file
Pasting passwords or secrets directly on the YAML seems non-ideal, because that file will possibly be commited on the Git repository, for example.

More ideal would be to read from a `GOBACKUP_ENC_PASSWORD`/`GOBACKUP_S3_ACCESS_KEY` environment variables, which could be set on the OS level or on a `.env` file. A separate `.env` file could be ignored in `.gitignore`, avoiding exposing the password.

### PR DESCRIPTION
**Pull request by ZachCheung**
```yaml
storages:
  r2:
    type: r2
    bucket: gobackup
    account_id: $R2_ACCOUNT_ID
    path: backups/demo
    access_key_id: ${R2_ACCESS_KEY_ID}
    secret_access_key: ${R2_SECRET_ACCESS_KEY}
```

`.env`:

```
R2_ACCOUNT_ID=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
R2_ACCESS_KEY_ID=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
R2_SECRET_ACCESS_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

```
2022/12/18 22:19:28 [Config] Load config from default path.
2022/12/18 22:19:28 [Model: demo] WorkDir: /tmp/gobackup3992736910/1671430768250469662/demo
2022/12/18 22:19:28 [Database] => database | sqlite: sqlite
2022/12/18 22:19:28 [SQLite] -> Dumping SQLite...
2022/12/18 22:19:28 [SQLite] dump path: /tmp/gobackup3992736910/1671430768250469662/demo/sqlite/sqlite/sample.sql
2022/12/18 22:19:28 [Database] Dump succeeded
2022/12/18 22:19:28 [Compressor] => Compress | xz
2022/12/18 22:19:28 [Compressor] -> /tmp/gobackup3992736910/1671430768250469662/2022.12.18.22.19.28.tar.xz
2022/12/18 22:19:28 [Storage] => Storage | r2
2022/12/18 22:19:28 [Cloudflare R2] -> Uploading (496 B)...
2022/12/18 22:19:28 [Cloudflare R2] => https://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.r2.cloudflarestorage.com/gobackup/backups/demo/2022.12.18.22.19.28.tar.xz
2022/12/18 22:19:28 [Cloudflare R2] Duration 628 milliseconds 789 microseconds, rate 1.0 MiB/s
2022/12/18 22:19:29 [Cycler] Removed 2022.12.18.22.17.23.tar.xz
2022/12/18 22:19:29 [Model] Cleanup temp: /tmp/gobackup3992736910/
2022/12/18 22:19:29 [Notifier] Running 0 Notifiers
```

### CODE REVIEW

**Comment by ZachCheung** [comment]
```yaml
storages:
  r2:
    type: r2
    bucket: gobackup
    account_id: $R2_ACCOUNT_ID
    path: backups/demo
    access_key_id: ${R2_ACCESS_KEY_ID}
    secret_access_key: ${R2_SECRET_ACCESS_KEY}
```

```shell
$ ls -hl .env
ls: cannot access '.env': No such file or directory
$ export R2_ACCOUNT_ID=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
$ export R2_ACCESS_KEY_ID=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
$ export R2_SECRET_ACCESS_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

```
2022/12/18 22:39:24 [Config] Load config from default path.
2022/12/18 22:39:24 [Model: demo] WorkDir: /tmp/gobackup2116670809/1671431964611685557/demo
2022/12/18 22:39:24 [Database] => database | sqlite: sqlite
2022/12/18 22:39:24 [SQLite] -> Dumping SQLite...
2022/12/18 22:39:24 [SQLite] dump path: /tmp/gobackup2116670809/1671431964611685557/demo/sqlite/sqlite/sample.sql
2022/12/18 22:39:24 [Database] Dump succeeded
2022/12/18 22:39:24 [Compressor] => Compress | xz
2022/12/18 22:39:24 [Compressor] -> /tmp/gobackup2116670809/1671431964611685557/2022.12.18.22.39.24.tar.xz
2022/12/18 22:39:24 [Storage] => Storage | r2
2022/12/18 22:39:24 [Cloudflare R2] -> Uploading (500 B)...
2022/12/18 22:39:25 [Cloudflare R2] => https://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.r2.cloudflarestorage.com/gobackup/backups/demo/2022.12.18.22.39.24.tar.xz
2022/12/18 22:39:25 [Cloudflare R2] Duration 632 milliseconds 799 microseconds, rate 1.0 MiB/s
2022/12/18 22:39:25 [Cycler] Removed 2022.12.18.22.34.45.tar.xz
2022/12/18 22:39:25 [Model] Cleanup temp: /tmp/gobackup2116670809/
2022/12/18 22:39:25 [Notifier] Running 0 Notifiers
```

File: config/config.go [hunk 1]
```diff
@@ -114,6 +116,21 @@ func Init(configFile string) {
...
+			logger.Errorf("Load %s failed: %v", dotEnv, err)
+			return
+		}
+
+		cfg, _ := os.ReadFile(viperConfigFile)
+		if err := viper.ReadConfig(strings.NewReader(os.ExpandEnv(string(cfg)))); err != nil {
+			logger.Errorf("Load expanded config failed: %v", err)
+			return
+		}
+	}
```
Comment (huacnlee) [inline config/config.go:132]: Need a integration test

**Comment by huacnlee** [comment]
`$R2_ACCOUNT_ID` and `${R2_ACCOUNT_ID}` variable supports is included in viper?

**Comment by ZachCheung** [comment]
> `$R2_ACCOUNT_ID` and `${R2_ACCOUNT_ID}` variable supports is included in viper?

Nope, it's expanded by `os.ExpandEnv()`.

**Pull request merged**